### ISSUE TITLE
Cannot access `content` property on result of octokit.repos.getContent()

### ISSUE THREAD

**Issue by wolfy1339** [octokit/rest.js#32]
The Typescript types for `octokit.repos.getContent` seem to be broken. 
Previously it would return a `Promise<OctokitResponse<ReposGetContentResponseData>>` which had a data property with a content property.  

Now i get TypeScript errors trying to get to **content** - either through .data or directly. 

```
 'content' does not exist on type '{ type: string; size: number; name: string; path: string; content?: string; sha: string; url: string; git_url: string; html_url: string; download_url: string; _links: { git: string; html: string; self: string; }; }[] | { ...; } | { ...; } | { ...; }'.
      Property 'content' does not exist on type '{ type: string; size: number; name: string; path: string; content?: string; sha: string; url: string; git_url: string; html_url: string; download_url: string; _links: { git: string; html: string; self: string; }; }[]'.
```

_Originally posted by @AndrewCsontos in https://github.com/octokit/rest.js/issues/1951#issuecomment-744449274_

**Comment by MarcCelani-at** [comment]
Is anyone still looking at this?

**Comment by wolfy1339** [comment]
Nothing can be done until it is resolved upstream.
Still blocked by: github/rest-api-description#165

**Comment by MarcCelani-at** [comment]
Thanks @wolfy1339 . Appreciate the context.

From our side, it appears that this issue has broken all type safety for rpc calls through this framework. I'd be happy to share code samples if that is helpful, but it seems from this discussion that there is wide acknowledgement that this is a broader issue that the content property on a single call. I'll add on that the way the rpcs are defined makes using an IDE really difficult. 

As a non-contributor to this project, it's not really my place to be too opinionated about the implementation details. I'm lacking a lot of the context about what the project's goals are or the history of how decisions were made. But as someone who has spent years of their life working on rpc frameworks, I would say that I consider type safety and good IDE integration to be quite possibly the most important features of an rpc library. I'm not sure if the issue is how my organization is using the rest package, but it appears to me like type safety is completely broken because of this issue. After a good holiday break, I would encourage the team to come back and really push on getting that upstream issue resolved, and if unable to succeed there, consider some outside the box options here, like... encapsulating the poorly typed code with a well typed interface, or even replacing the existing upstream dependency if important issues are not receiving enough attention. (My intent here is not to start a big flame war, but more to encourage the team to press forward on a solution and to find a way to get unblocked).

**Comment by wolfy1339** [comment]
Unfortunately, there isn't any active maintainer anymore.

You can read about it in [this discussion](https://github.com/octokit/octokit.js/discussions/620#discussioncomment-1472174). Make sure to subscribe to get any updates on the situation

The issue upstream lies with GitHub, it is up to them to fix it.

Our types are automatically derived from their OpenAPI spec, if there is a problem in their spec, it reflects in our types.

The methods in the library are also automatically generated from the OpenAPI spec. We don't define them manually.

There isn't any way to switch to an alternative upstream dependency as it is the source of truth.

Trust me when I say that if someone thought of  a solution it would have been implemented.

**Comment by wolfy1339** [comment]
I think we can add an override for this in [`octokit/openapi`](https://github.com/octokit/openapi) which would fix this until it's fixed upstream

**Comment by timrogers** [comment]
I have just created a PR within GitHub to fix the OpenAPI spec issue.

**Comment by timrogers** [comment]
@wolfy1339 I _think_ this should now be fixed. Are you able to give it a try?

**Comment by MarcCelani-at** [comment]
I'm still having issues. Consider the following:

```
const response = await octokit.orgs.checkMembershipForUser({
    org: organization,
    username,
});

// This is giving a ts error with "This condition will always return true, since '302' and '204' have no overlap.
// The type of response is OctokitResponse<unknown, 302>
if (response.status !== 204) {
    // Error handling code here
}
```

**Comment by wolfy1339** [comment]
That is a different issue. Please open another issue

**Comment by wolfy1339** [comment]
I used this snippet of code to test.
```ts
import { Octokit } from '@octokit/rest';
const octokit = new Octokit();

async function foo() {
  const { data } = await octokit.repos.getContent();
  if (!Array.isArray(data)) { // Filter out directories

    if (data.type === 'file') { // Make sure it's a file
      data.content;
    }
  }
}
```

> @wolfy1339 I *think* this should now be fixed. Are you able to give it a try?

Everything seems to work as expected now.

**Comment by oscard0m** [comment]
> github/rest-api-description#165

@wolfy1339 Should we give a try to [this proposal](https://github.com/octokit/rest.js/issues/32#issuecomment-759671901) now that https://github.com/github/rest-api-description/issues/165 has been solved?

**Comment by wolfy1339** [comment]
That should already be the case with the latest update. Your proposal was what was implemented in the OpenAPI spec.

**Comment by oscard0m** [comment]
> That should already be the case with the latest update. Your proposal was what was implemented in the OpenAPI spec.

Maybe the changes are not propagated yet to the rest of packages? [I drafted a TS Playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgeQMYwga2PAvnAMyghDgCIABCdLHAeigFMBnGMgbgChPUIA7VnGoZs8ALxw+jAO4oaogBQBKHv0ERpUqHAlkIzVAEMoAEwAMIMqoHwmkHeWmMARld424YQzAAWDsgBKAKIAggAiALJBAHQgJlbWgkzMkAKMDobShjhC8jjRyTAFjJDM0QDmjDAAwvwwjHwwCkgaWgA0cHYQHV6+cLhKXHDDI8OJ8O71jQ4AhCFQUIYAntHAzPOLSwrJqcyM0Sbehiqjp3AA-J0su-uHMIbRkw3wdHRwACpLYCyoUMBg8B8wHKPgANsCfDBmHBfGs4IY+HAQGtmMA+OV4TAYH9nABXepwaQ4Py+dIECCg0EaNEYxgLaBwZyMKnSThnUYALiuKTUtyOAG0zABdR51Z5cThAA), this snippet should be working, right?

**Comment by wolfy1339** [comment]
That snippet is incorrect as the `symlink` and `submodule` types do not have a `content` property, as said https://github.com/octokit/types.ts/issues/440#issuecomment-1220956680

**Comment by oscard0m** [comment]
> That snippet is incorrect as the `symlink` and `submodule` types do not have a `content` property, as said https://github.com/octokit/types.ts/issues/440#issuecomment-1220956680

Uops! I missed Tim's comment in that Issue.

Thanks and sorry for the noise here 🙏
